sympy/solvers/odesolvers/single.py

```diff
@@ -0,0 +1,243 @@
+#
+# This is the module for ODE solver classes for single ODEs.
+#
+
+from typing import ClassVar, Dict, List, Optional, Type
+
+from sympy.core.expr import Expr
+from sympy.core.function import AppliedUndef, Derivative, Function
+from sympy.core.relational import Equality
+from sympy.core.symbol import Symbol, Dummy
+from sympy.integrals import Integral
+from sympy.simplify import simplify
+
+from sympy.solvers.solvers import solve
+
+
+class ODEMatchError(NotImplementedError):
+    """Raised if a SingleODESolver is asked to solve an ODE it does not match"""
+    pass
+
+
+class SingleODEProblem:
+    """Represents an ordinary differential equation (ODE)
+
+    This class is used internally in the by dsolve and related
+    functions/classes so that properties of an ODE can be computed
+    efficiently.
+
+    Examples
+    ========
+
+    This class is used internally by dsolve. To instantiate an instance
+    directly first define an ODE problem:
+
+    >>> from sympy import Eq, Function, Symbol
+    >>> x = Symbol('x')
+    >>> f = Function('f')
+    >>> eq = f(x).diff(x, 2)
+
+    Noe you can create a SingleODEProblem instance and query its properties:
```

Comment:
```suggestion
    Now you can create a SingleODEProblem instance and query its properties:
```

Reply:
Thanks. I'll remember that when I rewrite this but I won't commit it now as there's no need to rerun the tests before rewriting.